src/app/project/list/list-item-form/edit-list-item/edit-list-item.component.ts

```diff
@@ -0,0 +1,113 @@
+import { Component, EventEmitter, Inject, Input, OnInit, Output } from '@angular/core';
+import { ApiResponseData, ApiResponseError, ChildNodeInfoResponse, KnoraApiConnection, List, ListNodeInfo, ListNodeInfoResponse, StringLiteral, UpdateChildNodeRequest } from '@dasch-swiss/dsp-js';
+import { DspApiConnectionToken } from '@dasch-swiss/dsp-ui';
+
+@Component({
+  selector: 'app-edit-list-item',
+  templateUrl: './edit-list-item.component.html',
+  styleUrls: ['./edit-list-item.component.scss']
+})
+export class EditListItemComponent implements OnInit {
+    loading: boolean;
+
+    @Input() iri: string;
+
+    @Input() projectIri: string;
+
+    @Output() closeDialog: EventEmitter<List | ListNodeInfo> = new EventEmitter<List>();
+
+    list: ListNodeInfo;
+
+    labels: StringLiteral[];
+    comments: StringLiteral[];
+
+    /**
+     * error checking on the following fields
+     */
+    formErrors = {
+        label: {
+            'required': 'A label is required.'
+        }
+    };
+
+    /**
+     * in case of an API error
+     */
+    errorMessage: any;
+
+    saveButtonDisabled = false;
+
+    formInvalidMessage: string;
+
+    constructor(@Inject(DspApiConnectionToken) private _dspApiConnection: KnoraApiConnection) { }
+
+    ngOnInit(): void {
+        this.loading = true;
+
+        // get list
+        this._dspApiConnection.admin.listsEndpoint.getListNodeInfo(this.iri).subscribe(
+            (response: ApiResponseData<ListNodeInfoResponse>) => {
+                this.list = response.body.nodeinfo;
+                this.buildForm(response.body.nodeinfo);
+            },
+            (error: ApiResponseError) => {
+                console.error(error);
+            }
+        );
+    }
+
+    buildForm(list: ListNodeInfo): void {
```

Comment:
would it be possible to add some description for each method, and possibly for tricky code lines?

Reply:
added in 2c88e7e